pkg/apis/app/v1alpha1/kogitotrusty_types.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KogitoTrustyCRDName is the name of the Kogito Trusty CRD in the cluster.
+const KogitoTrustyCRDName = "kogitotrusty.app.kiegroup.org"
+
+// KogitoTrustySpec defines the desired state of KogitoTrusty.
+// +k8s:openapi-gen=true
+type KogitoTrustySpec struct {
+	InfinispanMeta    `json:",inline"`
+	KafkaMeta         `json:",inline"`
+	KogitoServiceSpec `json:",inline"`
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+}
+
+// GetRuntime ...
+func (d *KogitoTrustySpec) GetRuntime() RuntimeType {
+	return QuarkusRuntimeType
+}
+
+// KogitoTrustyStatus defines the observed state of KogitoTrusty.
+// +k8s:openapi-gen=true
+type KogitoTrustyStatus struct {
+	KogitoServiceStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KogitoTrusty defines the Trusty Service infrastructure deployment.
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=kogitotrusty,scope=Namespaced
```

Comment:
```suggestion
// +kubebuilder:resource:path=kogitotrusties,scope=Namespaced
```

I believe you have to use in plural, that's why the CSV is not being generated right. Take a look in the CRDs that should be created by `operator-sdk generate` command to see the plural name for your type.